assets/app/view/part/town_rect.rb

```diff
@@ -147,10 +213,58 @@ def render
             stroke: 'none',
           })]
 
+        if (revenue = render_revenue)
+          children << revenue
+        end
+
         children << h(HitBox, click: -> { touch_node(@town) }, transform: translate) unless @town.solo?
 
         h(:g, children)
       end
+
+      def render_revenue
+        revenues = @town.revenue.values.uniq
+        return if revenues.uniq.size > 1
```

Comment:
this looks familiar again you already uniqued it

Reply:
9488efb